debian/entry.sh

```diff
@@ -58,7 +58,7 @@ function init_systemd()
 {
 	GREEN='\033[0;32m'
 	echo -e "${GREEN}Systemd init system enabled."
-	env > /etc/docker.env
+	env | sed 's/"/\\"/g' | sed 's/^\([a-zA-Z0-9_]*\)=\(.*\)$/\1="\2"/g' > /etc/docker.env
```

Comment:
Can we use shell escaping instead? I haven't tested if it works but if it does it will take care of all the edge cases.
```bash
for var in $(compgen -e); do
    printf '%q=%q\n' "$var" "${!var}"
done > /etc/docker.env
```